docs/source/testing.rst

```diff
@@ -0,0 +1,98 @@
+.. _`tox`: https://tox.readthedocs.io/en/latest/
+.. _`nise`: https://github.com/project-koku/nise
+
+~~~~~~~~~~~~
+Testing Koku
+~~~~~~~~~~~~
+
+Unit testing
+============
+
+Unit testing for Koku is handled through `tox`_.
+
+Running the test suite requires a minimum of an accessible database and Koku API server.
+
+Unit testing using docker-compose
+---------------------------------
+
+Example:
+::
+    $ make docker-up
+    $ tox
+    $ make docker-down
+
+Unit testing using OpenShift
+---------------------------------
+
+The ``Makefile`` provides convenience commands for enabling port-forwarding. This is required for accessing a database pod hosted within an OpenShift environment.
+
+Example:
+::
+    $ make oc-create-all
+    $ make oc-forward-ports
+    $ tox
+    $ make stop-forwarding-ports
+
+Smoke testing
+=============
+
+Running the QE functional and smoke tests is achieved through the scripts provided in ``koku.git/testing``
+
+Requirements
+------------
+In order to run the test suite, there need to be clones of the following git repositories:
+
+  - koku
+  - iqe
+  - hccm-plugin
+
+You will need to edit the settings file to match your environment: ``koku.git/testing/conf/settings.local.yaml``
```

Comment:
Similar question as above.  Are changes to this yaml needed for OpenShift local testing?  The way docker-compose is configured currently the user should never have to touch these files.

Reply:
Yes. You need to point it to the endpoint, which isn't going to be on localhost.